tests/Api/EncoderTest.php

```diff
@@ -102,53 +115,43 @@ public function testRun(): void
         if (function_exists('imagecreatefromwebp') && function_exists('imagecreatefromavif')) {
             $this->assertSame('image/webp', $this->getMime($this->encoder->setParams(['fm' => 'webp'])->run($this->avif)));
             $this->assertSame('image/avif', $this->getMime($this->encoder->setParams(['fm' => 'avif'])->run($this->webp)));
+            $this->assertSame('image/avif', $this->getMime($this->encoder->setParams(['fm' => 'avif'])->run($this->webpx)));
         }
     }
 
     public function testGetFormat(): void
     {
-        /**
-         * @psalm-suppress MissingClosureParamType
-         */
-        $image = \Mockery::mock(ImageInterface::class, function ($mock) {
-            /*
-             * @var Mock $mock
-             */
-            $this->assertMediaType($mock, 'image/jpeg')->once();
-            $this->assertMediaType($mock, 'image/png')->once();
-            $this->assertMediaType($mock, 'image/gif')->once();
-            $this->assertMediaType($mock, 'image/bmp')->once();
-            $this->assertMediaType($mock, 'image/jpeg')->twice();
-
-            if (function_exists('imagecreatefromwebp')) {
-                $this->assertMediaType($mock, 'image/webp')->once();
-            }
-
-            if (function_exists('imagecreatefromavif')) {
-                $this->assertMediaType($mock, 'image/avif')->once();
-            }
-        });
+        $this->assertSame('jpg', $this->encoder->setParams(['fm' => 'jpg'])->getFormat($this->getImageByMimeType('image/jpeg')));
```

Comment:
We also need tests where `fm` does not match the image's mime type. `fm`'s value should take precedence in that case.

Reply:
I added more tests to check the `fm` parameter always takes precedence